sdk/core/Azure.Core.Experimental/src/DynamicData.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+
+namespace Azure.Core.Dynamic
+{
+    /// <summary>
+    /// A dynamic abstraction over content data that may have an underlying
+    /// format of JSON or other format.
+    /// </summary>
+    public abstract class DynamicData
+    {
+        /// <summary>
+        /// Writes the data to the provided writer as a JSON value.
+        /// </summary>
+        /// <param name="writer">The writer to which to write the document.</param>
+        /// <param name="data">The dynamic data value to write.</param>
+        public static void WriteTo(Utf8JsonWriter writer, DynamicData data)
+        {
+            data.WriteTo(writer);
+        }
+
+        internal abstract void WriteTo(Utf8JsonWriter writer);
```

Comment:
Why is this internal?

Reply:
I didn't want it to be visible in the public API, because we are trying to make JsonData have no public APIs (with the exception of operator overloads) so everything must be accessed through the dynamic interface.

Reply:
If you have a specific concern, I'm happy to discuss, and will address in a later PR.